src/video/camerasource.cpp

```diff
@@ -214,14 +233,18 @@ void CameraSource::unsubscribe()
     }
     else
     {
+        //qDebug() << "second block";
         device->close();
         biglock = false;
     }
+    subscriptions--;
```

Comment:
There is also [cppcheck plugin](http://b.agilob.net/static-code-analysis-in-qtcreator-with-cppcheck/) that can be run on any platforms and integrates with g++. We fixed a lot issues in December using this.